engine/class_modules/warlock/sc_warlock_affliction.cpp

```diff
@@ -136,7 +136,10 @@ struct shadow_bolt_t : public affliction_spell_t
     if ( time_to_execute == 0_ms )
       p()->buffs.nightfall->decrement();
 
-    p()->buffs.decimating_bolt->decrement();
+    if ( rng().roll( p()->conduit.prolonged_decimation.percent() ) )
```

Comment:
Testing on Beta, I just noticed that the buff is currently consumed on impact, not execute. This is able to cause a buggy behavior, but we probably need to model this for now, so all these code sections in execute() regarding Decimating Bolt buff need to be moved unfortunately

Reply:
Update to this: For some reason Incinerate is on cast finish, so it's just this one and Demonbolt that need updating.

Reply:
Thanks for the input, I updated it accordingly now!